### ISSUE TITLE
Not possible to filter content overview by combining taxonomy and free text

### ISSUE THREAD

**Issue by hannesl** [bolt/bolt#7501]
We recently upgraded from 2.2 to 3.5. Our client is sad because they used to be able to search a content type (in the backend) by combining a category with a freetext search in such a way that the two limit the results – i.e. combined by an `AND`, not an `OR`. In other words, they want to be able to search for content in the "animals" category containing the word "dog". This was possible in 2.2 apparently.

I traced it down the stack to `Query::getContent()` with the following arguments:

```
[
    [0] => articles/search
    [1] => [
        [paging] => 1
        [hydrate] => 1
        [order] => -datepublish
        [filter] => dog
        [invisible] => 1
        [limit] => 10
        [categories] => animals
    ]
]

```

...which seems to make sense, but I would expect that `filter` and `categories` are combined with an `AND`.

Details
-------

 - Relevant Bolt Version: 3.5.1
 - Install type: Composer install
 - PHP version: 7.1
 - Used web server: Apache

Reproduction
------------

### Bug summary

In the overview of a content type in the backend (`/bolt/overview/{contenttype}`), when combining a *category* and a *text search* in the filter section, the combination of the two does not limit the result. In other words, the result contains posts from other categories than the one the user picks. This is a regression from 2.2.

### Steps to reproduce

1. Install Bolt
2. Add a content type that uses a taxonomy
3. Add some content using multiple taxonomy items. Make sure there is some text in the content that appears in multiple articles in different taxonomies.
3. Filter the overview with one taxonomy item and the text mentioned in 3).

### Expected result

The overview should only contain content that has the chosen taxonomy item.

### Actual result

Content containing other taxonomy items are returned, if they contain the text that was entered.

**Comment by rossriley** [comment]
I'll leave this open for others to add some input on. The current behaviour was designed to function as it is, the UI specifies that you are selecting a taxonomy OR a text filter and the code has been implemented to match this.

It does seem a little counter-intuitive to me since the actual need for doing an OR query here seems like it wouldn't really make sense, so if noone disagrees I'm happy to change the current implementation to do an AND query. Looping in @bobdenotter for UI input here.

**Comment by bobdenotter** [comment]
Yeah, let's keep this for now. I can see how it can be convenient for people to do an `AND` instead. :-)
